### ISSUE TITLE
[MU3 and MU4 Issue] Crash while converting to whole note

### ISSUE THREAD

**Issue by jason-cannon** [musescore/MuseScore#10653]
**Describe the bug**
A Crash to desktop occurs when selecting notes and converting them to whole notes twice. My guitar teacher was experiencing this issue and I am unsure if these are the steps he took exactly, but I reproduce the same CTD. I believe the issue has something to do with the mixture between notes and rhythm marks

**To Reproduce**
1. Open the attached muse score
2. Select the last measure with both notes and rythm marks
3. press 7 (convert to whole note) twice
4. CTD occurs

**Expected behavior**
Note should be converted to whole note and not crash

**Platform information**
 - OS: Windows

[Hero_of_War.zip](https://github.com/musescore/MuseScore/files/8121997/Hero_of_War.zip)

**Comment by Jojo-Schmitz** [comment]
I can't reproduce.

**Comment by Jojo-Schmitz** [comment]
Edit: I can (reading helps, press 7 twice)

**Comment by Jojo-Schmitz** [comment]
Possible fix:
```
diff --git a/libmscore/edit.cpp b/libmscore/edit.cpp
index 3461bb9503..b455f30137 100644
--- a/libmscore/edit.cpp
+++ b/libmscore/edit.cpp
@@ -4130,7 +4130,7 @@ static Element* findLinkedVoiceElement(Element* e, Staff* nstaff)
       Measure* measure = segment->measure();
       Measure* m       = score->tick2measure(measure->tick());
       Segment* s       = m->findSegment(segment->segmentType(), segment->tick());
-      return s->element(dtrack);
+      return s ? s->element(dtrack) : nullptr;
       }

 //---------------------------------------------------------
@@ -4169,6 +4169,8 @@ static Chord* findLinkedChord(Chord* c, Staff* nstaff)
       Segment* s  = c->segment();
       Measure* nm = nstaff->score()->tick2measure(s->tick());
       Segment* ns = nm->findSegment(s->segmentType(), s->tick());
+      if (!ns)
+            return 0;
       Element* ne = ns->element(dtrack);
       if (!ne->isChord())
             return 0;
@@ -4922,6 +4924,8 @@ void Score::undoAddElement(Element* element)
                         if (cr1->staffIdx() != cr2->staffIdx())
                               sm = cr2->staffIdx() - cr1->staffIdx();
                         Chord* c1 = findLinkedChord(cr1, score->staff(staffIdx));
+                        if (!c1)
+                              return;
                         Chord* c2 = findLinkedChord(cr2, score->staff(staffIdx + sm));
                         Note* nn1 = c1->findNote(n1->pitch(), n1->unisonIndex());
                         Note* nn2 = c2 ? c2->findNote(n2->pitch(), n2->unisonIndex()) : 0;
```

**Comment by Jojo-Schmitz** [comment]
I have added that fix to PR #9000

**Comment by Jojo-Schmitz** [comment]
It does apply, see #10654

**Comment by jason-cannon** [comment]
Wow, this was an incredibly fast turn around on this! Thanks @Jojo-Schmitz !

**Comment by Jojo-Schmitz** [comment]
As MuseSeoce 4 is affected by this too, can you remove that "[MU3]" from the title?

**Comment by Tantacrul** [comment]
I can get it to happen by selecting any measure with smaller notes and pressing 7 twice.

**Comment by Jojo-Schmitz** [comment]
Even/still with the artifact from PR #10654 ?

**Comment by Jojo-Schmitz** [comment]
It is available since more than 3 months now (and I've just rebased it now)...

**Comment by DmitryArefiev** [comment]
Fixed in #13183
